### PR TITLE
Rename the deferral config API to exception config API

### PR DESCRIPTION
**Pull request by md2119**
## Description

Renamed occurrences of `deferral` to `exception` in API and added validations as desired by the design. The affected API has not been released to the customers yet so no changelog required.

## Checklist
- [ ] Investigated and inspected CI test results
- [X] Unit test and regression tests added
- ~Evaluated and added CHANGELOG entry if required~
- ~Determined and documented upgrade steps~
- ~Documented user facing changes~

If any of these don't apply, please comment below.

## Testing Performed
Unit test

### CODE REVIEW

**Review by dvail — APPROVED** [review]

File: central/config/service/convert.go [hunk 1]
```diff
@@ -5,18 +5,18 @@ import (
 	"github.com/stackrox/rox/generated/storage"
 )
 
-// VulnerabilityDeferralConfigV1ToStorage returns a new instance of *storage.VulnerabilityDeferralConfig
-// based on input *v1.VulnerabilityDeferralConfig.
-func VulnerabilityDeferralConfigV1ToStorage(config *v1.VulnerabilityDeferralConfig) *storage.VulnerabilityDeferralConfig {
+// VulnerabilityDeferralConfigV1ToStorage returns a new instance of *storage.VulnerabilityExceptionConfig
+// based on input *v1.VulnerabilityExceptionConfig.
+func VulnerabilityDeferralConfigV1ToStorage(config *v1.VulnerabilityExceptionConfig) *storage.VulnerabilityExceptionConfig {
```
Comment (dashrews78) [inline central/config/service/convert.go:10]: nit:  Should update the func name and comment.

File: central/config/service/service.go [hunk 2]
```diff
@@ -118,6 +119,17 @@ func (s *serviceImpl) PutConfig(ctx context.Context, req *v1.PutConfigRequest) (
 	if req.GetConfig() == nil {
 		return nil, errors.Wrap(errox.InvalidArgs, "config must be specified")
 	}
+	if req.GetConfig().GetPrivateConfig() == nil {
```
Comment (dashrews78) [inline central/config/service/service.go:122]: nit:  Should we just move these nil checks into the `validateExceptionConfigReq`?

**Review by dashrews78 — APPROVED** [review]
2 nits.  I trust you will rename the convert function and completely your choice on whether to address the validation comment or not.

File: central/config/service/service.go (hunk 2 above)
Comment (md2119) [reply central/config/service/service.go:122]: This function receives the full config object which includes retention config, public config, etc. I was surprised to see that there is no validation done within this grpc function. I added a very basic check to ensure that at least private config is not nil to ensure that at least the zero value of the field is received.

**Pull request merged**